src/reducers/versions.tsx

```diff
@@ -624,6 +631,41 @@ const reducer: Reducer<VersionsState, ActionType<typeof actions>> = (
         },
       };
     }
+    case getType(actions.expandTree): {
+      const { versionId } = action.payload;
+
+      const version = state.versionInfo[versionId];
+      const { entries } = version;
+      const expandedPaths = entries
+        .filter((entry) => entry.type === 'directory')
```

Comment:
I didn't see any tests that mix directories and files. That would be a good test.